### ISSUE TITLE
Trying to understand fixations.x_hist vs. fixations.scanpath

### ISSUE THREAD

**Issue by nisar2** [matthias-k/pysaliency#78]
Hello,

I'm trying to understand the fixation.x_hist vs. fixations.scanpath. From my understanding, x_hist is a list of fixation history and has items of variable length. Essentially, you have a growing list "sandwiched" between empty lists. The item before the empty list represents a complete scanpath. I'm following the tutorial notebook and am working with MIT1003 so to test this understanding, I have the following block of code:

```python
for x in fixations.x_hist[:20]: print(x)
print('-'*100)
print(scanpaths.xs[0])
print(scanpaths.xs[1])
print(scanpaths.xs[2])
```

This gives me the following output:
![CleanShot 2024-06-25 at 11 54 58@2x](https://github.com/matthias-k/pysaliency/assets/70334112/ff833c39-627c-4a32-affe-e5d5bf339d54)

(I pasted the output as an image for the color coding). I expected the colored outputs to match based on my code. It seems like they match but the scanpaths have an extra item (highlighted in yellow). 

I'm wondering if my understanding of the data structure is wrong. I would love some clarity.

**Comment by matthias-k** [comment]
Hi @nisar2 ,

thanks for your question! I'll try to clarify the data structure a bit:

a `Fixations` instance contains many fixations which one might want to predict. Each fixation comes with information about its position (`x`, `y`) and information about the fixations which preceeded this fixation in the observer's scanpath (`x_hist`, `y_hist`). The most common case is that we're interested in all fixations for a list of scanpaths, this is what `ScanpathFixations(scanpaths)` is handling. In this case, the number of fixations is the sum of the lengths of all scanpaths.

In this case, `fixations.x` will iterate over the consecutive fixations of each scanpath (i.e., it will show _all_ fixations), while `x_hist` will contain for each fixation only the preceeding fixations. This is why it starts with an empty list for the first fixation in a scanpath, and ends with all but the last fixations for the last fixation in a scanpath.

The difference between `Scanpaths` and `Fixations` might indeed be a bit confusing and I have been thinking about unifying `Fixations` and `Scanpaths`, making `Fixations` a subclass of  `Scanpaths` where each scanpath has an history part and a last fixation.

TL;DR: `x_hist` contains only the fixations before the current fixation (the previous history). The last fixation in each scanpath is never part of any fixation's previous history.

Does this answer your question?

**Comment by nisar2** [comment]
Hi @matthias-k,

Thanks for the detailed response! Yes I think it makes sense. In other words, the empty lists in x.hist and y.hist indicate when a scanpath starts. 

Thank you!
